cleanlab_pinecone_rag_workflow/Cleanlab_Pinecone_RAG.ipynb

```diff
@@ -0,0 +1,2673 @@
+     "output_type": "stream",
+     "text": [
+      "Document: demand is there? How do you manage that? Is that perhaps by having a large proportion that you can a...\n",
+      "Metadata: {'chunk_id': '3d033601-ecd2-4d20-a69e-db85fcd35c3f', 'chunk_index': 122.0, 'filename': 'documents/Blackstone-Third-Quarter-2023-Investor-Call.pdf', 'quality': 'good', 'topic': 'finance', 'total_chunks': 127.0}\n",
+      "---\n",
+      "Document: %20 %12 %2 %5 %15 %—13 %Cover Page Cover PageSep. 29, 2022Cover [Abstract] Document Type Document Pe...\n",
+      "Metadata: {'chunk_id': '77a9db10-bd1c-46f9-b158-56a124d59205', 'chunk_index': 36.0, 'filename': 'documents/8k-nike.pdf', 'quality': 'good', 'topic': 'finance', 'total_chunks': 70.0}\n",
+      "---\n"
+     ]
+    }
+   ],
+   "source": [
+    "results = rag_pipeline.search(\n",
+    "    query=\"YOUR SEARCH QUERY HERE\",\n",
+    "    top_k=2,\n",
+    "    filter_query={\n",
+    "        \"topic\": {\"$eq\": \"finance\"},\n",
+    "        \"quality\": {\"$eq\": \"good\"}\n",
+    "    }\n",
+    ")\n",
+    "\n",
+    "for doc_text, metadata in results:\n",
+    "    print(f\"Document: {doc_text[:100]}...\")\n",
+    "    print(f\"Metadata: {metadata}\")\n",
+    "    print(\"---\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "160b19ad",
+   "metadata": {},
+   "source": [
+    "\n",
+    "Suppose our application asks this user which topic their question is about from a pre-defined list. Alternatively, we could train a classifier to predict the topic from the question (automatically using Cleanlab Studio AutoML).\n",
+    "\n",
+    "Below, you can specify the topic (or other metadata) when filtering to help search for the relevant content in your RAG index."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 174,
+   "id": "379aa3bf-c488-46ff-8c4a-6a52e6e49c89",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Top doc for this query: \n",
+      "\n",
+      "('Welcome to the topic on the sales process.In this topic, we will examine the basic steps in the sales process in SAP Business One. Then we will perform a streamlined sales process.2Imagine that your company has set customer satisfaction as its top priority. You review the basic steps in the sales process and decide to use a streamlined process for rush orders to ensure that customer demands can be met as quickly as possible.3(cid:132) The sales-A/R process moves', {'chunk_id': '6ddd27d2-7f72-451f-93a6-afd8f8c15907', 'chunk_index': 0.0, 'filename': 'documents/sap-sales-overview.pdf', 'quality': 'unknown', 'topic': 'sales', 'total_chunks': 15.0})\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Simple query regarding our documents\n",
+    "question = \"Tell me about the sales principles at SAP Business One\"\n",
+    "\n",
+    "top_doc = rag_pipeline.search(question, top_k=1, filter_query={\"topic\": {\"$eq\": \"sales\"}})\n",
```

Comment:
This example is duplicative to the one above (which filters on finance and good)?